beacon/goclient/goclient.go

```diff
@@ -187,6 +200,29 @@ func New(logger *zap.Logger, opt beaconprotocol.Options, operatorID spectypes.Op
 		zap.String("version", client.nodeVersion),
 	)
 
+	// If it's a Prysm, check that the debug endpoints are enabled, otherwise
+	// the Validators method might fail when calling BeaconState.
+	// TODO: remove this once Prysm enables debug endpoints by default.
+	if client.nodeClient == NodePrysm {
```

Comment:
just in terms of how it looks inside the `New` function, I would move that part to a separate function.

Reply:
done